src/components/legend/style.js

```diff
@@ -357,8 +357,30 @@ module.exports = function style(s, gd, legend) {
             var d0 = d[0];
             var w = boundLineWidth(d0.mlw, marker.line, MAX_MARKER_LINE_WIDTH, CST_MARKER_LINE_WIDTH);
 
-            p.style('stroke-width', w + 'px')
-                .call(Color.fill, d0.mc || marker.color);
+            p.style('stroke-width', w + 'px');
+
+            var fillColor = d0.mc || marker.color;
+
+            var getPatternAttr = function(mp, dflt) {
```

Comment:
This function looks quite similar to `getPatternAttr` in `src/components/drawing/index.js`.
Could you explain the difference between two?

Reply:
`getPatternAttr` in legends does not require any index because the first element of the array is always used, but the same `getPatternAttr` function can be used for both. If we unify them, which location do you think is appropriate for putting the common `getPatternAttr` fn?

Reply:
It could go in drawing.

Reply:
Sure! Fixed in 51614783013bdc48ea9d8076f44a31a8da95b5a5